osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -78,32 +82,51 @@ private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
                             Icon = FontAwesome.Solid.Star,
                             Colour = Color4.Black
                         },
-                        new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
+                        textFlow = new OsuTextFlowContainer(s => s.Font = OsuFont.Numeric.With(weight: FontWeight.Black))
                         {
                             Anchor = Anchor.CentreLeft,
                             Origin = Anchor.CentreLeft,
                             AutoSizeAxes = Axes.Both,
                             Direction = FillDirection.Horizontal,
                             TextAnchor = Anchor.BottomLeft,
-                        }.With(t =>
-                        {
-                            t.AddText($"{wholePart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 14);
-                                s.UseFullGlyphHeight = false;
-                            });
-
-                            t.AddText($"{separator}{fractionPart}", s =>
-                            {
-                                s.Colour = Color4.Black;
-                                s.Font = s.Font.With(size: 7);
-                                s.UseFullGlyphHeight = false;
-                            });
-                        })
+                        }
                     }
                 }
             };
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Current.BindValueChanged(difficulty => updateDisplay(difficulty.NewValue), true);
+        }
+
+        private void updateDisplay(StarDifficulty difficulty)
+        {
+            var starRatingParts = difficulty.Stars.ToString("0.00", CultureInfo.InvariantCulture).Split('.');
+            string wholePart = starRatingParts[0];
+            string fractionPart = starRatingParts[1];
+            string separator = CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator;
+
+            background.Colour = difficulty.DifficultyRating == DifficultyRating.ExpertPlus
+                ? ColourInfo.GradientVertical(Color4Extensions.FromHex("#C1C1C1"), Color4Extensions.FromHex("#595959"))
+                : (ColourInfo)colours.ForDifficultyRating(difficulty.DifficultyRating);
```

Comment:
not super sure about the lack of any sort of fade out transition here

Reply:
to clarify, you mean fading colours of the background?

Reply:
that, and the text sprites

Reply:
I've added proper transitioning along with rolling counter for the star rating display, I can imagine cases where the difficulty would need time to calculate before its shown, which would look bad on the player UX if it hides and shows back, moving the entire metadata flow back and forward. 

Not sure on the design of the placeholder though.

https://user-images.githubusercontent.com/22781491/117552234-c4c53f00-b052-11eb-9d42-2d7db1a5e5cf.mp4

Reply:
I'm not sure why this was added in this PR, but it's weird due to the font being right aligned and not fixed width.

Reply:
Please split this out into its own PR and fix that issue there. The bubble / text can't shift every frame of the transition.

Reply:
Using fixed-width made it look really weird so I refrained from doing that, and I thought it wasn't that big of a change to split it, but can do.

Reply:
Well it's affecting a global component and you've updated it in a local usage. There's nothing to say you actually checked how this works in every other usage.

I'm not saying it should be fixed width, but if anything it should not be right aligned and should definitely not affect the size of the parent container to the point it's bouncing around.

Reply:
I'm not really sure about how to fix this, I've checked fixed-width one more time to record a video of it and the results are indeed not appealing: (especially with `1.23` due to font being right-aligned as you mentioned)

https://user-images.githubusercontent.com/22781491/117606549-caaa4580-b162-11eb-8914-b3d5a930397c.mp4

Also fixing the width of the star rating display to 60px makes it look a bit questionable:

https://user-images.githubusercontent.com/22781491/117607349-756f3380-b164-11eb-898a-21e9ecb4067b.mp4

The `StarRatingDisplay` in general looks to be designed to not be changeable and make the numbers roll around, not sure how to proceed forward without leaving this as-is (or going with the second option of fixing width to 60px).

Reply:
Fixed width looks okay I think. If it's decided it doesn't then the numbers should not roll in the first place.

Reply:
It shouldn't update all the time, only when the difficulty calculation is heavy, so I think it's fine to remove the rolling altogether and call it a day for this design.